core.js

```diff
@@ -978,6 +978,22 @@ exports.init = function (sbot, config) {
 
   const reindexingLock = mutexify()
 
+  function LazyPull() {
+    let onValue
+
+    return {
+      add(val) {
+        if (onValue) onValue(null, val)
+      },
+      source(abort, cb) {
+        if (abort) cb(abort)
+        else if (!onValue) onValue = cb
+      },
+    }
+  }
+
+  const decryptedValues = LazyPull()
```

Comment:
This seems to do the same as `pull-notify`. Could we use that instead? Because it is unit tested and because it evokes a familiar pattern, may helps developers understand what's going on here.

Reply:
right, that was the module I was looking for :)